### DISCUSSION TITLE
publishing cloned npm package with collaboration mode

### DISCUSSION THREAD

**Discussion by akashg3627** [excalidraw/excalidraw#3941] [Q&A]
First of all, thanks for creating such a good tool
I have been trying to use collaboration mode of excalidraw in my project where I want to use collaboration mode and save the canvas as image to my server on button click. As mentioned in discussion #3879 , I am using the 4 steps mentioned by @ad1992 to clone the code and create my own npm package. I could achieve 3 steps but facing some errors while using the npm package.
I have published the package as @akashg3627/excalidraw-colab-test 
I am getting the following error while using the package

![Screenshot (60)](https://user-images.githubusercontent.com/67226124/130852539-4ba7e68c-5fcb-435c-af17-a2fc6a8f3e64.png)
![Screenshot (61)](https://user-images.githubusercontent.com/67226124/130852545-c6ecf063-82d6-4744-abbd-b669de718568.png)

It would be great if you can please guide me further and help me solve this error.

**Comment by ad1992** [comment]
@akashg3627 do you have `window.EXCALIDRAW_ASSET_PATH` specified before loading excalidraw ? because by default it takes from `unpkg` cdn but as per the error its taking from root so looks like somewhere you have defined it. If you remove that this error should get fixed. If you want to load it from the root path you will need to serve the assets from the path.

**Comment by akashg3627** [comment]
Can you please help me out with this issue, I have tried removing window.EXCALIDRAW_ASSET_PATH from index.html too when using excalidraw-app/index for collab mode, also there were some error due to env file that has to be solved by DefinePlugin in web.config files which actually gave me the result for a moment and I could use collab in my local app using the package I built after modification. But on reload it vanishes of giving the same error of missing localhost/exclaidraw-asset. 
And if there is someone who did this earlier, please kindly let me know about it.

**Reply by ad1992** [reply]
What is the entry file in webpack ? You need to make sure you import "./publicPath" in your entry file at the top so all chunks use `__webpack_public_path__`

**Reply by akashg3627** [reply]
Thanks a lot, it worked and I was able to implement my requirements

**Reply by ad1992** [reply]
Awesome 👍

**Comment by anupvarma** [comment]
Looking for something similar approach bundling excalidraw-app as a plugin and using it for collaboration purpose in an existing react application. @ad1992 @akashg3627 do you have any example for this please ?
